ch2driver/pytpcc/drivers/awss3driver.py

```diff
@@ -0,0 +1,178 @@
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+# -----------------------------------------------------------------------
+
+from __future__ import with_statement
+
+import os
+import sys
+import logging
+#import pymongo
+from pprint import pprint,pformat
+#from pymongo.mongo_client import MongoClient
+#from pymongo.server_api import ServerApi
+import traceback
+import time
+import constants
+import boto3
+import uuid
+import json
+import io
+import csv
+from .abstractdriver import *
+
+## ==============================================
+## Awss3Driver
+## ==============================================
+class Awss3Driver(AbstractDriver):
+    DEFAULT_CONFIG = {
+        "host":         ("The hostname to mongod", "localhost" ),
+        "port":         ("The port number to mongod", 27017 ),
+        "name":         ("Collection name", "tpcc"),
+        "denormalize":  ("If set to true, then the CUSTOMER data will be denormalized into a single document", False),
+    }
+
+    def __init__(self, ddl, clientId, TAFlag="T",
+                 schema=constants.CH2_DRIVER_SCHEMA["CH2"],
+                 preparedTransactionQueries={},
+                 analyticalQueries=constants.CH2_DRIVER_ANALYTICAL_QUERIES["HAND_OPTIMIZED_QUERIES"],
+                 customerExtraFields=constants.CH2_CUSTOMER_EXTRA_FIELDS["NOT_SET"],
+                 ordersExtraFields=constants.CH2_ORDERS_EXTRA_FIELDS["NOT_SET"],
+                 itemExtraFields=constants.CH2_ITEM_EXTRA_FIELDS["NOT_SET"],
+                 load_mode=constants.CH2_DRIVER_LOAD_MODE["NOT_SET"],
+                 load_format=constants.CH2_DRIVER_LOAD_FORMAT["JSON"],
+                 kv_timeout=constants.CH2_DRIVER_KV_TIMEOUT,
+                 bulkload_batch_size=constants.CH2_DRIVER_BULKLOAD_BATCH_SIZE):
+        super(Awss3Driver, self).__init__("awss3", ddl)
+        try:
+            self.client_id = clientId;
+            self.TAFlag = TAFlag
+            self.schema = schema
+            self.s3client = boto3.client('s3')
+            if self.client_id == 0 and self.TAFlag == "L":
+                self.s3client.create_bucket(Bucket=self.schema)
```

Comment:
S3 bucket names need to be [globally unique](https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html) so naming the bucket the same as the schema can cause problems. I think we should let the user optionally provide a bucket name, and if they don't then the bucket name needs to have some UUID component to it (e.g. take the last N digits of a UUID).

Also, I think it would be good to explicitly provide the AWS region when creating `s3client` and when creating the bucket. The user could specify this with a CLI flag, and it could have a default of `us-east-1` for example. Right now, it seems like you need to have `AWS_DEFAULT_REGION` configured as an environment variable for it to work (or at least, for me it wasn't working before I set this env var), which isn't super obvious.

Reply:
WIll defer these to the next iteration.